client/src/context/PreferencesContext.tsx

```diff
@@ -48,11 +52,14 @@ export const PreferencesProvider = ({ children }: PreferencesProviderProps) => {
 
     loadPreferences();
   }, []);
-
+  const { changeLanguage, currentLanguage } = useLocales();
```

Comment:
better to place all declarations at the start of the function. For eg; in line 41